bin/cli

```diff
@@ -1,20 +1,39 @@
 #!/usr/bin/env node
 
 const kitVersion = require('../package.json').version
-const [majorNodeVersion] = process.version.split('.')
-const [majorKitVersion] = kitVersion.split('.')
-if (majorNodeVersion !== 'v16' && majorNodeVersion !== 'v18') {
-  console.warn('')
-  console.warn('\nYou\'re using Node', process.version, 'but GOV.UK Prototype Kit', `v${majorKitVersion}`, 'only supports node versions 16 and 18.\n')
-  console.warn('Please upgrade to node 18 and try again.\n')
-  console.warn('')
+const { parse } = require('./utils/argv-parser')
+
+const argv = parse(process.argv, {
+  booleans: ['no-version-control', 'verbose', 'running-within-create-script', 'use-njk-extensions']
+})
+
+const majorNodeVersion = parseInt(process.versions.node.split('.')[0], 10)
+
+if (majorNodeVersion !== 16 && majorNodeVersion !== 18) {
+  const isStage2Migrate = argv.command === 'migrate' && process.argv[3] === '--'
+  const isStage2Create = argv.command === 'init'
+
+  // Do not display error or warning messages when this is stage2 of create or migrate
```

Comment:
What's the reason for making exceptions for "stage 2"? I'm nervous about this because it puts the argument parser into a scope where it needs to run on node versions which we don't support.  Anything above the `process.exit` line needs to be much more widely compatible than the rest of our codebase.

Reply:
This was to prevent the warning message being displayed twice.  If you have a better way to prevent this, it would be great.

Reply:
I'd like to use a flag to suppress the message - we can check for that with simpler code which will be more backwards compatible and less likely to break in the future as there is no `require` before the version check.  For example this line works in the latest version of node v4: `!!process.argv.find(x => x === '--suppress-node-version-warning' )`.

The argv parser is likely to get more complex and I don't think it's good to have to stick to older syntax in that file.